jupyter_server/base/handlers.py

```diff
@@ -251,9 +251,28 @@ def identity_provider(self):
             )
             from jupyter_server.auth import IdentityProvider
 
-            # no identity provider set, load default
+            non_alphanum = re.compile(r"[^A-Za-z0-9]")
+            default_cookie_name = non_alphanum.sub("-", f"username-{self.request.host}")
+
+            # If there is no identity provider set, load the default. If using
+            # a classic notebook server, adding extensions that inherit
+            # from JupyterHandler will use a mix of new+old authentication log.
+            # Here, we construct an identity provider that works side-by-side
+            # and consistently with the old way that we handled auth in
+            # the classic server.
             self.settings["identity_provider"] = IdentityProvider(
-                config=self.settings.get("config", None)
+                config=self.settings.get("config", None),
+                # For backwards compatibility, pass the token
+                # from the webapp settings.
+                token=self.settings.get("token", "<generated>"),
```

Comment:
If the token's not set, this will actually set the token to the value "<generated>" and accept requests authenticated with that string.